platform/qemu-virt/rules.mk

```diff
@@ -37,7 +37,8 @@ MODULE_DEPS += \
 GLOBAL_DEFINES += \
     MEMBASE=$(MEMBASE) \
     MEMSIZE=$(MEMSIZE) \
-    PLATFORM_SUPPORTS_PANIC_SHELL=1
+    PLATFORM_SUPPORTS_PANIC_SHELL=1 \
+	CONSOLE_HAS_INPUT_BUFFER=1
```

Comment:
Did you verify that this all works with the input buffer turned off again? This platform/target should probably be a good test case since it has code in both paths.